repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -311,14 +349,66 @@ class Interpreter(prompt0: Ref[String],
       (classFiles, newImports) <- compiled
       _ = Timer("cachedCompileBlock 4")
       cls <- eval.loadClass(fullyQualifiedName, classFiles)
-    } yield (cls, newImports)
+    } yield (cls, newImports, tag)
+
   }
 
-  def processModule(source: ImportHook.Source, code: String, wrapperName: Name, pkgName: Seq[Name]) = {
 
-    processModule0(source, code, wrapperName, pkgName, predefImports)
+  def evalCachedClassFiles(cachedData: Seq[CompileCache], pkg: String, wrapper: String): Unit = {
+
+    def evalMain(cls: Class[_]) =
```

Comment:
This is already a method in the `Evaluator` class; perhaps we can move it out into the `Evaluator` companion object so this method can all it? Rather than copy-pasting